internal/trace/transport.go

```diff
@@ -88,14 +88,14 @@ func logHeader(header http.Header, e log.Logger) {
 
 // SetHTTPDebugLog sets up http debug log with logrus.Logger
 func SetHTTPDebugLog(ctx context.Context, authClient *auth.Client) {
-	if logrusLog, ok := log.GetLogger(ctx).(*logrus.Logger); ok && logrusLog.Level != logrus.DebugLevel {
+	if logrusLog, ok := log.GetLogger(ctx).(*logrus.Logger); !ok || logrusLog.Level != logrus.DebugLevel {
 		return
 	}
 	if authClient.Client == nil {
-		authClient.Client = http.DefaultClient
+		authClient.Client = &http.Client{}
 	}
 	if authClient.Client.Transport == nil {
-		authClient.Client.Transport = http.DefaultTransport
+		authClient.Client.Transport = http.DefaultTransport.(*http.Transport).Clone()
```

Comment:
The original one looks good as we are not changing anything in `http.DefaultTransport` (even at line 100). It is not necessary to clone a new transport as long as we have new instance at line 95.
```suggestion
		authClient.Client.Transport = http.DefaultTransport
```